webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { Spinner } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { Table } from '../common/table';
+import { STATUS } from '../../constants';
+import MessageBox from '../common/MessageBox';
+import { translate as __ } from '../../common/I18n';
+import createModelsTableSchema from './ModelsTableSchema';
+import { getURIQuery } from '../../common/helpers';
+import Pagination from '../Pagination/Pagination';
+
+class ModelsTable extends React.Component {
+  state = { initialLoading: true };
+  componentDidMount() {
+    this.props.getTableItems('models', getURIQuery(window.location.href));
+  }
+
+  stopSpinner() {
+    this.setState({ initialLoading: false });
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      this.state.initialLoading &&
+      prevProps.status === STATUS.PENDING &&
+      this.props !== STATUS.PENDING
+    ) {
+      this.stopSpinner();
+    }
+  }
+
+  render() {
+    const {
+      getTableItems,
+      sortBy,
+      sortOrder,
+      error,
+      status,
+      results,
+      data: { pagination },
+    } = this.props;
+
+    const renderTable =
+      status === STATUS.ERROR ? (
+        <MessageBox
+          key="models-table-error"
+          icontype="error-circle-o"
+          msg={__(`Could not receive data: ${error && error.message}`)}
+        />
+      ) : (
+        <React.Fragment>
+          <Table
+            key="models-table"
+            columns={createModelsTableSchema(getTableItems, sortBy, sortOrder)}
+            rows={results}
+          />
+          <div id="pagination">
+            <Pagination data={pagination} />
+          </div>
+        </React.Fragment>
+      );
+
+    return (
+      <Spinner size="lg" loading={this.state.initialLoading}>
```

Comment:
I have mixed feelings about this.
I think its good to show a spinner for the initial loading, but it also shows when changing pages,
the page flickers and the table disappears. 

IMO, you should check the state, if the results array is empty, show a spinner, if not show the table. this way, there won't be any flickers, and I think you can remove the internal state of the component along with the `componentDidUpdate` and `stopSpinner`
```suggestion
    if (results.length > 0) return renderTable;
    return <Spinner size="lg" loading />;
```